dart/lib/flat_buffers.dart

```diff
@@ -1170,20 +1171,36 @@ class Uint16Reader extends Reader<int> {
   int read(BufferContext bc, int offset) => bc._getUint16(offset);
 }
 
-/// Reader of lists of unsigned 8-bit integer values.
-///
-/// The returned unmodifiable lists lazily read values on access.
+/// Reader of unmodifiable binary data (a list of unsigned 8-bit integers).
 class Uint8ListReader extends Reader<List<int>> {
-  const Uint8ListReader();
+  /// Enables lazy reading of the list
+  ///
+  /// If true, the returned unmodifiable list lazily reads bytes on access.
+  /// Therefore, the underlying buffer must not change while accessing the list.
+  ///
+  /// If false, reads the whole list immediately as an Uint8List.
+  final bool lazy;
+
+  const Uint8ListReader({this.lazy = true});
```

Comment:
I don't follow... there's a single constructor... what sorting?

Reply:
Put constructor first in the class, rather than instance variables first

Reply:
hm, where does that convention come from? All other classes (in flat_buffers.dart) have instance variables first

Reply:
Ahh ok. https://dart-lang.github.io/linter/lints/sort_constructors_first.html is the lint for it - we can go back and apply it universally later rather than introduce different conventions in this PR. I'm more used to Flutter style, which uses that lint.

Reply:
OK